parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh

```diff
@@ -80,6 +80,75 @@ cleanUpGPUDrivers() {
     rm -Rf $GPU_DEST /opt/gpu
 }
 
+installContainerd() {
+    packageVersion="${3:-}"
+    containerdMajorMinorPatchVersion="$(echo "$packageVersion" | cut -d- -f1)"
+    containerdHotFixVersion="$(echo "$packageVersion" | cut -d- -f2)"
+    CONTAINERD_DOWNLOADS_DIR="${1:-$CONTAINERD_DOWNLOADS_DIR}"
+    eval containerdOverrideDownloadURL="${2:-}"
+
+    # the user-defined package URL is always picked first, and the other options won't be tried when this one fails
+    if [[ ! -z ${containerdOverrideDownloadURL} ]]; then
+        installContainerdFromOverride ${containerdOverrideDownloadURL} || exit $ERR_CONTAINERD_INSTALL_TIMEOUT
+        return 0
+    fi
+    installContainerdWithAptGet "${containerdMajorMinorPatchVersion}" "${containerdHotFixVersion}" "${CONTAINERD_DOWNLOADS_DIR}" || exit $ERR_CONTAINERD_INSTALL_TIMEOUT
+}
+
+installContainerdFromOverride() {
+    containerdOverrideDownloadURL=$1
+    echo "Installing containerd from user input: ${containerdOverrideDownloadURL}"
+    # we'll use a user-defined containerd package to install containerd even though it's the same version as
+    # the one already installed on the node considering the source is built by the user for hotfix or test
+    logs_to_events "AKS.CSE.installContainerRuntime.removeMoby" removeMoby
+    logs_to_events "AKS.CSE.installContainerRuntime.removeContainerd" removeContainerd
+    logs_to_events "AKS.CSE.installContainerRuntime.downloadContainerdFromURL" downloadContainerdFromURL "${containerdOverrideDownloadURL}"
+    logs_to_events "AKS.CSE.installContainerRuntime.installDebPackageFromFile" "installDebPackageFromFile ${CONTAINERD_DEB_FILE}" || exit $ERR_CONTAINERD_INSTALL_TIMEOUT
+    echo "Succeeded to install containerd from user input: ${containerdOverrideDownloadURL}"
+    return 0
+}
+
+installContainerdWithAptGet() {
```

Comment:
The logic of this function is mostly from the original func `installStandaloneContainerd`